app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -326,6 +326,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
         }
 
+        settings().topSitesSize = components.core.topSitesStorage.cachedTopSites.size
```

Comment:
Since the `TopSiteFeature` handles fetching the list of top sites and adding them to `cachedTopSites` we don't have a callback AFAIK and I wasn't sure how to handle this otherwise without making changes in AC. Any ideas?

Reply:
This reminds me of how we emitted facts to tell how many web extensions were installed for the metric pings. See https://github.com/mozilla-mobile/android-components/pull/5998. However, there was a callback for initializing the web extensions compare to the TopSiteFeature. 

I think one way of doing this is to emit a fact with the size when we call `getTopSites` in [DefaultTopSitesStorage](https://github.com/mozilla-mobile/android-components/blob/be232773113848eb2ebd270b17271448423aee46/components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt#L89), which we can follow up with setting the `topSitesSize` in Fenix. 

I would recommend making the change in AC because I think it is following a better practice of isolating the logic to their component for reusability despite the extra work required, and it also helps potential consumers of AC beyond Fenix (which I think would be really good for mobile).